cityofaustin/atd-vz-data
# Updates crash mode trigger, socrata python script

## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/12065

## Testing
**URL to test:** 
you can test this on [staging](https://visionzero-staging.austinmobility.io/editor/#/crashes)! i have run all the scripts in this pr on staging, with the exception of the python script (to update the socrata open data portal dataset that powers the viewer), which should deploy when this pr is merged to main.

this pr adds new mode categories and includes a one-time script to update the crashes table's mode category metadata column to reflect changes made to units records (i ran this on staging as well, and we will need to run it on prod before the next release), and a trigger which will automate this process going forward. this should result in more accurate mode data in the vision zero viewer and will make it possible to represent micromobility devices and e-scooters in the viewer

we will need a separate issue (if there isn't one already) to make sure socrata is updating properly and make sure vzv can represent the new categories. thanks y'all!

**Steps to test:**
- open a record in the crashes table on staging and the [hasura staging console](https://vzd-staging.austinmobility.io/console/api-explorer) in a separate tab
- change the 'type' on one of them to be 'micromobility device' and the body style to 'unknown'
- query the record in the hasura console's graphiql interface by running the following query and replacing the crash id with the current record
```
query MyQuery {
  atd_txdot_crashes(where: {crash_id: {_eq: 18900579}}) {
    atd_mode_category_metadata
    units {
      atd_mode_category
      body_style {
        veh_body_styl_desc
        veh_body_styl_id
      }
      unit_desc_id
      unit_description {
        veh_unit_desc_desc
        veh_unit_desc_id
      }
      unit_id
    }
  }
}
```
- the `atd_txdot_crashes` `atd_mode_category_metadata` column for the updated record should show 'micromobility device' (this should allow the vz team to account for micromobility devices that are not scooters)
- change the body style to 'e-scooter' and rerun the query
- the metadata column should show 'e-scooter'
- change the type to 'motor vehicle' and body style to 'police motorcycle' and rerun the query
- the metadata column should show 'motorcycle' (these were previously not being counted as motorcycles)
- change the type to 'motorized conveyance' and body style to 'unknown' and rerun the query
- the metadata column should show 'motorized conveyance'
- try out other combinations and rerun the query to confirm the metadata matches the unit record
---
#### Ship list
